### PR TITLE
satellite: rearrange marketing package

### PR DESCRIPTION
**Pull request by VinozzZ**
What: 
- Remove `marketing package`
- Create `Offers` package

Why:

- We need to expose offer table to the `console` package and `marketing` package. It's not idea to tie `marketing` package with `console` package. 
- A satellite operator should be able to choose which marketing campaign they would like to run in stead of only have a high-level marketing package
- The `value attribution` has its own package for open source program so `marketing` package should not exists

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/offers/offers.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package offers
```
Comment (egonelbre) [inline satellite/offers/offers.go]: Is there a better name than `offers`?

e.g. `kickback`, `rewards`...?

Comment (VinozzZ) [reply satellite/offers/offers.go]: I do like `rewards`!

Comment (egonelbre) [reply satellite/offers/offers.go]: I suspect there will eventually be some other logic related to `rewards`...

with regards to package naming:
* https://golang.org/doc/effective_go.html#package-names
* https://rakyll.org/style-packages/
* https://github.com/golang/go/wiki/CodeReviewComments#package-names
* https://blog.golang.org/package-names

`rewardsconfig` seems overly specific. The package responsibility is to organize logic about `rewards`.

File: satellite/satellitedb/offers.go [hunk 2]
```diff
@@ -10,26 +10,26 @@ import (
...
 
-// ListAll returns all offers from the db
-func (offers *offers) ListAll(ctx context.Context) ([]marketing.Offer, error) {
-	offersDbx, err := offers.db.All_Offer(ctx)
+// ListAll returns all offersDB from the db
+func (db *offersDB) ListAll(ctx context.Context) ([]offers.Offer, error) {
+	offersDbx, err := db.db.All_Offer(ctx)
 	if err != nil {
-		return nil, marketing.OffersErr.Wrap(err)
+		return nil, offers.Err.Wrap(err)
```
Comment (egonelbre) [inline satellite/satellitedb/offers.go]: Note the wrapping error should be from this package not the imported one.

File: satellite/offers/offers.go (hunk 1 above)
Comment (VinozzZ) [reply satellite/offers/offers.go]: That sounds reasonable. I changed it

**Comment by VinozzZ** [comment]
The only reason I don’t want to change the struct type name right now is
that I want to keep it consistent with the table name which is offers.
We also use the term free credit offer. Either way, I think we can make
these changes for now and if we do see a big confusion or advantage of
changing all naming to rewards. We can do it in another PR

File: internal/testplanet/satellite.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 	"storj.io/storj/satellite/console"
 	"storj.io/storj/satellite/console/consoleweb"
 	"storj.io/storj/satellite/mailservice"
-	"storj.io/storj/satellite/marketing/marketingweb"
```
Comment (coyle) [inline internal/testplanet/satellite.go]: curious, why the change ?

File: satellite/peer.go [hunk 4]
```diff
@@ -605,6 +605,7 @@ func New(log *zap.Logger, full *identity.FullIdentity, db DB, config *Config, ve
 		peer.Marketing.Endpoint, err = marketingweb.NewServer(
 			peer.Log.Named("marketing:endpoint"),
 			marketingConfig,
+			// peer.DB.Offers()
```
Comment (coyle) [inline satellite/peer.go]: should this still be here ?

File: satellite/satellitedb/offers.go [hunk 5]
```diff
@@ -10,26 +10,31 @@ import (
...
 	"github.com/zeebo/errs"
 
-	"storj.io/storj/satellite/marketing"
+	"storj.io/storj/satellite/rewards"
 	dbx "storj.io/storj/satellite/satellitedb/dbx"
 )
 
-type offers struct {
+var (
+	// Error the default offers errs class
```
Comment (coyle) [inline satellite/satellitedb/offers.go]: should be in the form of `offerErr is the default offer errors class`

File: internal/testplanet/satellite.go (hunk 3 above)
Comment (VinozzZ) [reply internal/testplanet/satellite.go]: Yea, from the conversation we had last week about the package structure, the service layer is not needed since both `console` and `marketingweb` server needs access to the `offers` table. It feels weird to put `offers` under marketing only.

File: satellite/peer.go (hunk 4 above)
Comment (VinozzZ) [reply satellite/peer.go]: Yes, I will remove this
